solvebio/resource/dataset.py

```diff
@@ -24,58 +26,148 @@ class Dataset(CreateableAPIResource,
     Datasets are access points to data. Dataset names are unique
     within a vault folder.
     """
-    USES_V2_ENDPOINT = True
+    RESOURCE_VERSION = 2
 
     LIST_FIELDS = (
         ('id', 'ID'),
-        ('vault_object_name', 'Name'),
+        ('vault_name', 'Vault Name'),
+        ('vault_object_path', 'Vault Object Path'),
+        ('vault_object_filename', 'Vault Object Filename'),
+        ('created_at', 'Created'),
         ('description', 'Description'),
     )
 
     @classmethod
-    def get_by_full_path(cls, full_path, **kwargs):
-        from solvebio import Object
+    def make_full_path(cls, vault_name, path, name):
         from solvebio import SolveError
 
         try:
-            obj = Object.retrieve_by_full_path(full_path)
-            dataset = Dataset.retrieve(obj['dataset_id'], **kwargs)
-            return dataset
+            user = client.get('/v1/user', {})
+            domain = user['account']['domain']
         except SolveError as e:
-            if e.status_code != 404:
-                raise e
+            print("Error obtaining account domain: {0}".format(e))
+            raise
+
+        result = ':'.join([
+            domain,
+            vault_name,
+            os.path.join('/' + path.lstrip('/'), name),
+        ])
+
+        return result
 
     @classmethod
-    def create_by_name(cls, name, vault_name, path, **kwargs):
+    def get_by_full_path(cls, full_path, **kwargs):
+        from solvebio import Object
+
+        obj = Object.retrieve_by_full_path(full_path)
+        dataset = Dataset.retrieve(obj['dataset_id'], **kwargs)
+        return dataset
+
+    @classmethod
+    def get_or_create_by_name(cls, vault_name, path, name, **kwargs):
         from solvebio import Vault
         from solvebio import Object
         from solvebio import SolveError
 
+        create_vault = kwargs.pop('create_vault')
+
+        path = re.sub('//+', '/', path)
+
+        if path != '/':
+            path.rstrip('/')
+
+        try:
+            test_path = Dataset.make_full_path(vault_name, path, name)
+        except SolveError as e:
+            print("Error obtaining account domain: {0}".format(e))
+            raise
+
+        try:
+            dataset = Dataset.get_by_full_path(test_path)
+
+            # If the dataset exists but the genome_builds don't match,
+            # update it with the new builds.
+            if dataset.is_genomic and \
+                    dataset.genome_builds != kwargs.get('genome_builds'):
+                dataset.genome_builds = kwargs.get('genome_builds')
+                dataset.save()
```

Comment:
I don't really know.  I just copied it from what is presently there:

https://github.com/solvebio/solvebio-python/blob/master/solvebio/resource/dataset.py#L40

Does something need to change with this?